core/src/main/java/org/elasticsearch/index/search/MatchQuery.java

```diff
@@ -355,6 +355,9 @@ protected Query blendTermQuery(Term term, MappedFieldType fieldType) {
                     if (query instanceof FuzzyQuery) {
                         QueryParsers.setRewriteMethod((FuzzyQuery) query, fuzzyRewriteMethod);
                     }
+                    if (query == null) {
```

Comment:
I think we catch it if it happens with the assertion no? so we are handling the null values at the source, plus we cover the case where we missed something. that should be enough.